docs/node_management_status.md

```diff
@@ -0,0 +1,117 @@
+        * `certVersion`: The version of the certificate to be installed.
+        * `configVersion`: The version of the config to be installed.
+    * `status`: The state, of the upgrade job. See the section **Status Values** below for more information.
+    * `k8s`: A JSON structure to further define the upgrade progress of a cluster agent.
+        * `configMap`: A JSON structure to track the upgrade progress of the config map.
+            * `needChange`: A boolean value that indicates if the config map needs to be updated.
+            * `updated`: A boolean value that indicates if the config map was already updated by the node management worker.
+        * `secret`: 
+            * `needChange`: A boolean value that indicates if the secret needs to be updated.
+            * `updated`: A boolean value that indicates if the secret was already updated by the node management worker.
+        * `imageVersion`: 
+            * `needChange`: A boolean value that indicates if the agent deployment needs to be upgraded or downgraded.
+            * `updated`: A boolean value that indicates if the agent deployment was already upgraded or downgraded by the node management worker.
+            * `from`: The version of the agent deployment before execution of the agent upgrade job.
+            * `to`: The version that the agent upgrade job will attempt to upgrade or downgrade the agent deployment to.
+    * `errorMessage`: A short message that describes why an agent upgrade job has failed.
+    * `workingDirectory`: The directory that the upgrade job will be reading and writing files to.
+
+## Status values
+
+* Agent Auto Upgrade status values
+    * `"waiting"`: The node management worker has matched the NMP to this node and created the status object in the local database.
+    * `"download started"`: The download worker has began downloading all necessary packages from the Management Hub.
+    * `"downloaded"`: The download worker has finished downloading all necessary packages from the Management Hub.
+    * `"initiated"`: The installation of the downloaded packages has started and is being performed by the AgentAutoUpgrade cron job script.
+    * `"successful"`: The node management worker has successfully performed the upgrade job specified in the NMP.
+    * `"no action required"`: The node management worker has determined that no actions need to be taken to upgrade or downgrade the agent. This typicaly means that all the files specified within the NMP's manifest are already installed, or they are a lower version than what is currently installed, and the NMP set the allowDowngrade field to false.
+    * `"download failed"`: The download worker was unable to download all necessary packages from the Management Hub.
+    * `"failed"`: There was a problem during the installation of the downloaded packages either in the node management worker or in the AgentAutoUpgrade cron job script.
```

Comment:
One more status added recently: 'upgrade aborted'. This is when the pre-check failed before the real upgrade happens.